examples/conv_lstm.py

```diff
@@ -8,11 +8,18 @@
 from keras.layers.normalization import BatchNormalization
 import numpy as np
 import pylab as plt
+from keras import backend as K
 
 # We create a layer which take as input movies of shape
 # (n_frames, width, height, channels) and returns a movie
 # of identical shape.
 
+
+if K.backend() == 'cntk':
+    raise RuntimeError('CNTK could not run this example, due to the missing support for padding '
+                       'with non-specified input shape. To run it with CNTK, please update the '
+                       'input_shape in top COnvLSTM2D layer from (None, 40, 40, 1) to (15, 40, 40, 1)')
```

Comment:
For your concern of too many "if" into keras code. we already tried to wrapper most changes in cntk backend, to avoid add "if" to the keras layer code. The only exception is the local layer, because the way cntk support it is totally different... If keras has strong concern about that part, we can have more detail discussion (like try to support the keras way in the next CNTK Keras GA release). If you have concern for other parts, please let me know.